chain/x/zoracle/handler.go

```diff
@@ -215,10 +216,28 @@ func handleMsgRequestData(ctx sdk.Context, keeper Keeper, msg MsgRequestData) sd
 }
 
 func handleMsgReportData(ctx sdk.Context, keeper Keeper, msg MsgReportData) sdk.Result {
+	startGas := ctx.GasMeter().GasConsumed()
+
+	// Save new report to store
 	err := keeper.AddReport(ctx, msg.RequestID, msg.DataSet, msg.Sender)
 	if err != nil {
 		return err.Result()
 	}
+
+	// Calculate total refunded by multiply RefundGasPrice with gas used
+	refundedAmount, _ := msg.RefundGasPrice.MulDec(sdk.NewDec(int64(ctx.GasMeter().GasConsumed() - startGas))).TruncateDecimal()
```

Comment:
@taobun . Should we use `GasConsumed` or `GasConsumedToLimit`? also above

Reply:
No difference here because if gas consumed exceed the limit, it will panic out of gas before this line.